Support TTL for EmbeddedRocksDB (#39986)
ClickHouse/ClickHouse: docs/en/engines/table-engines/integrations/embedded-rocksdb.md

    name1 [type1] [DEFAULT|MATERIALIZED|ALIAS expr1],
    name2 [type2] [DEFAULT|MATERIALIZED|ALIAS expr2],
    ...
) ENGINE = EmbeddedRocksDB([ttl, read_only]) PRIMARY KEY(primary_key_name)
```

Engine parameters:

> - `ttl` - time to live. TTL is accepted in seconds. Not specifying/passing or non-positive TTL behaves like TTL = infinity.

Reviewer: ```suggestion
- `ttl` - time to live for values. TTL is accepted in seconds. If TTL is 0 or negative behaves, regular RocksDB instance is used (without TTL).
```

Author: `ttl` is `Int32`, so a negative number is also acceptable here

Reviewer: But we don't care about negative numbers only 0 or positive

> - `read_only` - when `read_only` be set `true` opens in the usual read-only mode. Compactions will not be triggered(neither manual nor automatic), so no expired entries removed.

Reviewer: This is not a TTL only feature, we can use it for non-ttl database also.
Let's change the description to

```suggestion
- `read_only` - when `read_only` is set to `true`, read-only mode is used. For storage with TTL, compaction will not be triggered (neither manual nor automatic), so no expired entries are removed.
```
